lib/floe/workflow/choice_rule.rb

```diff
@@ -4,24 +4,28 @@ module Floe
   class Workflow
     class ChoiceRule
       class << self
-        def true?(payload, context, input)
-          build(payload).true?(context, input)
-        end
-
         def build(payload)
```

Comment:
This is 100% NOT for this PR but it got me thinking we might want to add some validation here that the payload is valid per the rules defined in the spec [[ref]](https://states-language.net/#choice-state)

E.g.
> Boolean expression
>
> A Boolean Expression is a JSON object which contains a field named "And", "Or", or "Not". If the field name is "And" or "Or", the value MUST be an non-empty object array of Choice Rules, which MUST NOT contain "Next" fields; the interpreter processes the array elements in order, performing the boolean evaluations in the expected fashion, ceasing array processing when the boolean value has been unambiguously determined.
>
> The value of a Boolean Expression containing a "Not" field MUST be a single Choice Rule, that MUST NOT contain a "Next" field; it returns the inverse of the boolean to which the Choice Rule evaluates.

Reply:
yea. I was thinking the constructors could ensure that a next or end were defined among others.

and the boolean expression parsing could do the same

Reply:
Yeah I'd like to raise `Floe::InvalidWorkflowError` for things like this on initial load not at runtime so parsing this at the start is great